src/uct/api/uct_def.h

```diff
@@ -210,6 +210,176 @@ typedef struct uct_cm_remote_data {
+     * Fields not specified by this mask will be ignored.
+     */
+    uint64_t                   field_mask;
+
+    /**
+     * Indicates the client's status.
+     */
+    ucs_status_t               status;
+} uct_cm_ep_server_connect_cb_handle_t;
+
+
+/**
+ * @ingroup UCT_CLIENT_SERVER
+ * @brief Client endpoint's connect callback arguments field mask.
+ *
+ * The enumeration allows specifying which fields in
+ * @ref uct_cm_ep_client_connect_cb_handle are present, for backward compatibility
+ * support.
+ */
+enum uct_cm_ep_client_connect_cb_handle_field {
+    /** Enables @ref uct_cm_ep_client_connect_cb_handle::remote_data */
+    UCT_CM_EP_CLIENT_CONNECT_CB_HANDLE_REMOTE_DATA = UCS_BIT(0),
+
+    /** Enables @ref uct_cm_ep_client_connect_cb_handle::status */
+    UCT_CM_EP_CLIENT_CONNECT_CB_HANDLE_STATUS      = UCS_BIT(1)
+};
+
+
+/**
+ * @ingroup UCT_CLIENT_SERVER
+ * @brief Arguments to the client's connect callback.
+ *
+ * Used with the client-server API on a connection manager.
+ */
+typedef struct uct_cm_ep_client_connect_cb_handle {
+    /**
+     * Mask of valid fields in this structure, using bits from
+     * @ref uct_cm_ep_server_connect_cb_handle_field.
+     * Fields not specified by this mask will be ignored.
+     */
+    uint64_t                   field_mask;
+
+    /**
+     * Remote data from the server.
+     */
+    const uct_cm_remote_data_t *remote_data;
+
+    /**
+     * Indicates the server's status.
+     */
+    ucs_status_t               status;
+} uct_cm_ep_client_connect_cb_handle_t;
+
+
+/**
+ * @ingroup UCT_CLIENT_SERVER
+ * @brief Listener's connection request callback arguments field mask.
+ *
+ * The enumeration allows specifying which fields in
+ * @ref uct_cm_listener_conn_req_cb_handle are present, for backward compatibility
+ * support.
+ */
+enum uct_cm_listener_conn_req_cb_handle_field {
+    /** Enables @ref uct_cm_listener_conn_req_cb_handle::local_dev_name */
+    UCT_CM_LISTENER_CONN_REQ_CB_HANDLE_LOCAL_DEV_NAME = UCS_BIT(0),
+
+    /** Enables @ref uct_cm_listener_conn_req_cb_handle::conn_request */
+    UCT_CM_LISTENER_CONN_REQ_CB_HANDLE_CONN_REQUEST   = UCS_BIT(1),
+
+    /** Enables @ref uct_cm_listener_conn_req_cb_handle::remote_data */
+    UCT_CM_LISTENER_CONN_REQ_CB_HANDLE_REMOTE_DATA    = UCS_BIT(2)
+};
+
+
+/**
+ * @ingroup UCT_CLIENT_SERVER
+ * @brief Arguments to the listener's connection request callback.
+ *
+ * The local device name, connection request handle and the data the client sent.
+ * Used with the client-server API on a connection manager.
+ */
+typedef struct uct_cm_listener_conn_req_cb_handle {
+    /**
+     * Mask of valid fields in this structure, using bits from
+     * @ref uct_cm_listener_conn_req_cb_handle_field.
+     * Fields not specified by this mask will be ignored.
+     */
+    uint64_t                   field_mask;
+
+    /**
+     * Device name which handles the incoming connection request.
+     */
+    const char                 *local_dev_name;
```

Comment:
let's use "dev_name[UCT_DEVICE_NAME_MAX]" to be consistent with uct_tl_device_resource_t
also rename to "dev_name" for shortness

Reply:
i prefer to keep 'local' for better understanding of the code later on, ok?

Reply:
the device name is always local, we never pass device names of remote side